### PR TITLE
CSHARP-4070: Fix flaky Ensure_command_network_error_before_hadnshake_is_correctly_handled test.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: tests/MongoDB.Driver.Core.Tests/Jira/CSharp3173Tests.cs [hunk 1]
```diff
@@ -326,27 +342,28 @@ private void ForceClusterId(MultiServerCluster cluster, ClusterId clusterId)
             void SetupFailedConnection(Mock<IConnection> mockFaultyConnection)
             {
                 // async path is not used in serverMonitor
-                var faultyConnectionResponses = new Queue<Action>(new Action[]
+                var faultyConnectionResponses = new ConcurrentQueue<Action>(new Action[]
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Core.Tests/Jira/CSharp3173Tests.cs:345]: Is `ConcurrentQueue` needed here, as it looks like the action queue is local to the connection?

Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Core.Tests/Jira/CSharp3173Tests.cs]: true, done

File: tests/MongoDB.Driver.Core.Tests/Jira/CSharp3173Tests.cs [hunk 2]
```diff
@@ -326,27 +342,28 @@ private void ForceClusterId(MultiServerCluster cluster, ClusterId clusterId)
...
                     });
 
                 mockFaultyConnection
                     .Setup(c => c.ReceiveMessage(It.IsAny<int>(), It.IsAny<IMessageEncoderSelector>(), It.IsAny<MessageEncoderSettings>(), It.IsAny<CancellationToken>()))
                     .Returns(() =>
                     {
-                        // wait until the command network error has been triggered
-                        WaitForTaskOrTimeout(hasNetworkErrorBeenTriggered.Task, TimeSpan.FromMinutes(1), "network error");
+                        WaitForTaskOrTimeout(
+                            Ensure.IsNotNull(hasNetworkErrorBeenTriggered, nameof(hasNetworkErrorBeenTriggered)).Task,
```
[the remark this answers is not in the document]
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Core.Tests/Jira/CSharp3173Tests.cs]: done

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**